doc/source/train/user-guides/experiment-tracking.rst

```diff
@@ -1,195 +1,364 @@
+.. tabs::
 
-        Example:
+    .. tab:: Wandb
+
+        **online**
+
+        Ensure that credentials are set inside of ``train_func``.
 
         .. code-block:: python
             
-            import pytorch_lightning as pl
-            from pytorch_lightning.loggers.wandb import WandbLogger
-            from pytorch_lightning.loggers.comet import CometLogger
-            from pytorch_lightning.loggers.mlflow import MLFlowLogger
-
-            def train_func_per_worker():
-                ...
-
-                wandb_logger = WandbLogger(
-                    name="demo-run", 
-                    project="demo-project", 
-                    id="unique_id",  
-                    offline=offline
-                )
-                
-                comet_logger = CometLogger(
-                    api_key=YOUR_COMET_API_KEY,
-                    experiment_name="demo-experiment",
-                    project_name="demo-project,
-                    offline=offline,
-                )
-                
-                mlflow_logger = MLFlowLogger(
-                    run_name=name,
-                    experiment_name=project_name,
-                    tracking_uri=f"file:{save_dir}/mlflow",
-                )
-                
-                trainer = pl.Trainer(
-                    # ...,
-                    logger=[wandb_logger, comet_logger, mlflow_logger],
-                )
-
-        .. tip::
-
-            Always make sure to set necessary credentials on each worker.
-            Ray Train will not automatically port environment varaibles 
-            on the head node to workers.
-
-            Taking Wandb as an example, you can manually set `WANDB_API_KEY`
-            on each worker. This allows the ``WandbLogger`` to fetch credentials 
-            from environment variables and automatically login to your Wandb account.
-
-            .. code-block:: python
-                :emphasize-lines: 5,9
-                
-                import os
-                from pytorch_lightning.loggers.wandb import WandbLogger
-
-                # Grab the api key from the head node
-                WANDB_API_KEY = os.environ["WANDB_API_KEY"]
-
-                def train_func_per_worker():
-                    # Set environment variable on each worker
-                    os.environ["WANDB_API_KEY"] = WANDB_API_KEY
-                    wandb_logger = WandbLogger(...)
-                    ...
-
-                trainer = TorchTrainer(
-                    train_func_per_worker,
-                    ...
-                )
+            # This is equivalent to `os.environ["WANDB_API_KEY"] = "your_api_key"`
+            wandb.login(key="your_api_key")
+
+        **offline**
+
+        Ensure that offline directory points to a shared storage path.
+
+        .. code-block:: python
+
+            os.environ["WANDB_MODE"] = "offline"
+            wandb.init(dir="some_shared_storage_path/wandb") 
```

Comment:
Can you define Online and Offline directly in here? For online, this means using a W&B API Key to authenticate to the serve, and log to W&B server. For offline, this means no authentication and logging directly to shared file system as the backend.


This would
1: Make it clear how this maps to the next subsections (setting up credentials, setting up shared storage)
2. Allow you to use MLflow terminology (local, remote) rather than online/offline.